python_modules/dagster/dagster/_loggers/__init__.py

```diff
@@ -93,7 +99,27 @@ def json_logged_job():
 
     class JsonFormatter(logging.Formatter):
         def format(self, record):
-            return _seven.json.dumps(record.__dict__)
+            dict_to_dump = {}
+            for k, v in record.__dict__.items():
+                if k == LOG_RECORD_EVENT_ATTR:
+                    # Redundant with the "dagster_event" field under "dagster_meta"
+                    continue
+                elif k == LOG_RECORD_METADATA_ATTR:
+                    # Events objects are not always JSON-serializable, so need to pack them first
+                    json_serializable_event = pack_value(v[LOG_RECORD_EVENT_ATTR])
+                    json_serializable_dagster_meta = DagsterLogRecordMetadata(
+                        dagster_event=json_serializable_event,
+                        **{
+                            meta_k: meta_v
+                            for meta_k, meta_v in v.items()
+                            if meta_k != "dagster_event"
+                        },
+                    )
+                    dict_to_dump[LOG_RECORD_METADATA_ATTR] = json_serializable_dagster_meta
```

Comment:
IIUC above conditional branch can be written considerably shorter (clearer too IMO):

```
        dict_to_dump[LOG_RECORD_METADATA_ATTR] = DagsterLogRecordMetadata({
            **v, "dagster_event": pack_value(v[LOG_RECORD_EVENT_ATTR])
        }
```

Reply:
Ah yup - rewrote